tests/unit/dynamodb/test_conditions.py

```diff
@@ -284,6 +284,10 @@ def test_eq(self):
             },
         )
 
+    def test_eq_repr(self):
+        actual_repr = str(Equals(self.value, self.value2))
+        assert actual_repr == 'mykey = foo'
```

Comment:
@tim-finnigan / @tibbe I could use your input on these tests before I flesh them out fully. One option is what I'm doing here and adding a bunch of `test_op_repr` tests. Another option is to change `ConditionBase#get_expression` so that it returns a new `repr` key with the string representation and then update the existing `test_op` test cases to assert on the `repr` also.

I don't have a preference either way but let me know what you would prefer.

Reply:
actually I already fleshed them out but happy to revise based on comments. otherwise I think this PR is ready for review.

Reply:
Sorry for the late reply. We might need to distinguish between what `__str__` and `__repr__` does. According to the [specification of `__repr__`](https://docs.python.org/3/reference/datamodel.html#object.__repr__):

> If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value (given an appropriate environment). If this is not possible, a string of the form `<...some useful description...>` should be returned. The return value must be a string object. If a class defines [`__repr__()`](https://docs.python.org/3/reference/datamodel.html#object.__repr__) but not [`__str__()`](https://docs.python.org/3/reference/datamodel.html#object.__str__), then [`__repr__()`](https://docs.python.org/3/reference/datamodel.html#object.__repr__) is also used when an “informal” string representation of instances of that class is required.

In the case of `__repr__` we probably want something that looks like constructor application e.g. `And(AttributeExists("x"), AttributeExists("y"))`. For `__str__` we could then have whatever pretty-prints nicely e.g. the syntax actually used in [DynamoDB query expressions](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.ConditionExpressions.html).

Reply:
I had originally started with `__str__` but was asked to use `__repr__` instead here:
https://github.com/boto/boto3/pull/3254#issuecomment-1121465272

tbh I'm not particular as long as we can get this PR merged and have human understandable representations so that debugging is a nicer experience.

@tim-finnigan what can I do to help get this PR merged?

Reply:
If we go with a textual description the string should be surrounded by `<>`, as per the documentation of `__report__()`, or otherwise it will be hard to parse when there is surrounding text.

In this case I think we could actually get a real `__repr__()` (i.e. one fulfilling "If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value (given an appropriate environment).") by instead implementing `__repr__()` along the lines of:

```py
def __repr__(self):
  return f'{self.__class__.__name__}({", ".join(self._values)})'
```